### ISSUE TITLE
unique mash-up and covers?

### ISSUE THREAD

**Issue by Mister-G-Lu** [richard512/findyourmusic#2]
Would you consider adding cool covers, for example, major to minor (where Chase Holfelder's "Beauty and the Beast" distinguishes itself from mere dark pop like Worst in Me by Pluto, due to merely different tone to lyrics of a different author), or mash-ups, for example Rather Be is quite normal classical cross-over pop, September is pretty normal disco music, but then Rather Be in September -- suddenly becomes unique and strange; classical-cross over disco is very unusual and never heard of in normal music.

**Comment by richard512** [comment]
Yes! Any appealing contributions like this: I'm up for adding it in.

**Comment by Mister-G-Lu** [comment]
nice! Also, what do you think about music mostly meant for humor? The lip-sync for Star Wars called "Eagles (stop it now)" is very silly, but it is actually pretty decent to listen to. I Am Potato is a weird song depending on randomish lyrics (if giraffe will fly then cookie dough will melt???!) yet has an undeniable catchiness to it.

**Comment by richard512** [comment]
Absolutely everything. Fork the repo and please feel free to push any genres you'd like to add. As soon as I see the contributions, I'll add them to the project.

**Comment by Mister-G-Lu** [comment]
How do I view all 1,500+ files to ensure I have no repeats?

**Comment by richard512** [comment]
Make sure you've got git installed on your machine, then download a copy of your fork of the repo using this command:

`git clone https://github.com/Mister-GoodwinLu/FindYourMusic.git`

That'll make a directory called FindYourMusic which contains all of the files in the project.

You can then take a look at any and all files in the "songs" directory.

Next, you can add mp3s of genres you want to add.

To add them to the FindYourMusic playlist you can add the songs to the "music" array mentioned on [line 88 of index.html](https://github.com/Mister-GoodwinLu/FindYourMusic/blob/master/index.html#L88)

You may prefer to do something like auto-detection of the mp3 files in the songs directory via a PHP script instead of manually entering the filenames in the "music" array.

When you're happy with the additions and changes, you can run these commands to update your copy of the repo:

```
git commit -a

git push
```

Then it'll ask you for your github login info and upload the changes to your copy of the repo.

Next, you can do a pull request [here](https://github.com/Mister-GoodwinLu/FindYourMusic/pulls) in order to add the updates you've made to the original FindYourMusic repo.

I probably said things you already know, but I hope this was useful.

**Comment by Mister-G-Lu** [comment]
do I just download the mp3 to my own computer? Or find a website using an mp3 of the song?
And finally, what quality do you consider your examples of genres to be? For example, your Rap has "crazy brazy", however, the vast majority of people would probably pin Lose Yourself as the masterpiece of rap songs due to its prowess and message. Do you just throw in a random song for any genre, or do you attempt to showcase the very utmost top song?

**Comment by richard512** [comment]
They're often popular songs within each genre, but I'm open to the idea of alternative songs that may be more iconic of a given genre. In order to add a song you would probably want to download a song to your computer, chop a 30 second sample from it, and add it to the songs directory.

I just made index.php and added it to the project. It auto-detects and adds mp3 files in the songs directory. You can run it by having PHP installed on your computer and running this command in the FindYourMusic directory:

`php -S localhost:1337`

That starts up a temporary PHP web server on port 1337 which serves up index.php at http://localhost:1337 -- That way, you can just add mp3s and they'll automatically show up in the web app.

**Comment by Mister-G-Lu** [comment]
how would I "chop a 30 second sample"?

**Comment by richard512** [comment]
Probably something like Audacity or ffmpeg.

**Comment by Mister-G-Lu** [comment]
what do you think about unique artists in their genres? For example, Pink Floyd is obviously Rock, but his music sounds completely different from any other rock artist. Similarly, Lindsey Stirling is EDM, but EDM hardly ever involves fancy violin playing.

**Comment by richard512** [comment]
Route 1: Base the selections on popularity within each genre
Route 2: Base the selctions on in-depth consideration of the elements of the genre
I'd guess Route 1 would be what connects best with most people, through Route 2 would probably be better for people who really dig deep into the music.    On Thursday, June 7, 2018, 2:24:10 PM PDT, Goodwin Lu <notifications@github.com> wrote:  
 
 
what do you think about unique artists in their genres? For example, Pink Floyd is obviously Rock, but his music sounds completely different from any other rock artist. Similarly, Lindsey Stirling is EDM, but EDM hardly ever involves fancy violin playing.
